test/functional/apps/visualize/_point_series_options.js

```diff
@@ -203,11 +204,87 @@ export default function ({ getService, getPageObjects }) {
 
         const labels = await PageObjects.visualize.getXAxisLabels();
 
-        await kibanaServer.uiSettings.replace({ 'dateFormat:tz': 'Browser' });
+        expect(labels).to.eql(expectedLabels);
+      });
+
+      it('should show different labels in different timezone', async function () {
+        const fromTime = '2015-09-22 09:05:47.415';
+        const toTime = '2015-09-22 16:08:34.554';
+        // note that we're setting the absolute time range while we're in 'America/Phoenix' tz
+        await PageObjects.timePicker.setAbsoluteRange(fromTime, toTime);
+        await PageObjects.common.sleep(10000);
+        const labels = await PageObjects.visualize.getXAxisLabels();
+        log.debug(`Labels: ${labels}`);
+        expect(labels).to.eql([ '10:00', '11:00', '12:00', '13:00', '14:00', '15:00' ]);
+
+        const expectedTableData = [ [ '09:05', '13', '13,463,070,562.462' ],
+          [ '09:10', '23', '11,518,321,384.727' ],
+          [ '09:15', '24', '12,437,509,461.333' ],
+          [ '09:20', '30', '14,439,976,253.793' ],
+          [ '09:25', '23', '11,017,524,802.783' ],
+          [ '09:30', '30', '10,774,443,820.138' ],
+          [ '09:35', '25', '10,565,619,548.16' ],
+          [ '09:40', '21', '16,412,910,738.286' ],
+          [ '09:45', '21', '13,207,024,435.2' ],
+          [ '09:50', '26', '12,091,266,626.783' ],
+          [ '09:55', '11', '8,882,773,271.273' ],
+          [ '10:00', '17', '15,367,929,856' ],
+          [ '10:05', '17', '10,990,063,375.059' ],
+          [ '10:10', '11', '12,884,901,888' ],
+          [ '10:15', '14', '10,200,547,328' ],
+          [ '10:20', '21', '12,240,656,793.6' ],
+          [ '10:25', '10', '10,737,418,240' ],
+          [ '10:30', '12', '10,111,068,842.667' ],
+          [ '10:35', '10', '11,381,663,334.4' ],
+          [ '10:40', '14', '11,657,768,374.857' ]
+        ];
+        await inspector.open();
+        await inspector.expectTableData(expectedTableData);
+        await inspector.close();
+        log.debug('set \'dateFormat:tz\': \'UTC\'');
+        await kibanaServer.uiSettings.replace({ 'dateFormat:tz': 'UTC', 'defaultIndex': 'logstash-*' });
+        // We set the tz from 'America/Phoenix' to UTC and refreshing the browser but not re-entering
+        // the absolute time range so the timepicker is going to shift +7 hours.
         await browser.refresh();
+        // wait some time before trying to check for rendering count
+        await PageObjects.header.awaitKibanaChrome();
+        await PageObjects.visualize.waitForRenderingCount();
```

Comment:
Based on the implementation it looks like we could also use `waitForVisualizationRenderingStabilized` here, right?

Reply:
If we are 100% sure that there is only 1 viz rendering to be completed - `waitForRenderingCount` is a better choice. Sometimes we cannot garantee the fixed count and then we have to use `waitForVisualizationRenderingStabilized`

Reply:
I'm now using `retry.waitFor` since even waitForRenderCount doesn't always wait for the axis to be rendered.